src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/AwsSdk/SQSRequestHandler.cs

```diff
@@ -107,8 +107,11 @@ public static AfterWrappedMethodDelegate HandleSQSRequest(InstrumentedMethodCall
                         var ec = executionContext;
                         var response = ec.ResponseContext.Response; // response is a ReceiveMessageResponse
 
-                        // accept distributed trace headers from the first message in the response
-                        SqsHelper.AcceptDistributedTraceHeaders(transaction, response.Messages[0].MessageAttributes);
+                        if (response.Messages != null && response.Messages.Count > 0 && response.Messages[0].MessageAttributes != null)
```

Comment:
Could we refactor this block and the one just like it below into a separate method? Or maybe move the logic to inspect `response.Messages` into the `SqsHelper.AcceptDistributedTraceHeaders()` method?

Reply:
Good idea, done.